search/collector.go

```diff
@@ -30,3 +30,20 @@ type Collector interface {
 	SetFacetsBuilder(facetsBuilder *FacetsBuilder)
 	FacetResults() FacetResults
 }
+
+
+type DocumentMatchHandler func(hit *DocumentMatch) error
+
+type MakeDocMatchHandlerKey string
```

Comment:
sometimes is called DocumentMatch, sometimes DocMatch?  Recommend be consistent on DocumentMatch?